### PR TITLE
Qt5 port of Tweetian on Sailfish

### PR DESCRIPTION
**Pull request by siteshwar**
implemented basic functionality for Qt5 port of Tweetian.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -16,15 +16,16 @@
     along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include <QtGui/QApplication>
-#include <QtDeclarative/QDeclarativeContext>
-#include <QtDeclarative/QDeclarativeView>
-#include <QtDeclarative/qdeclarative.h>
+#include <QtWidgets/QApplication>
```
Comment (veskuh) [inline main.cpp:19]: Would be better be without QWidget dependencies. The main application could be QGuiApplication

File: rpm/tweetian.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Requires:
+- sailfishsilica-qt5
+- qt5-qtdeclarative-import-location
+- qt5-qtdeclarative-import-positioning
+- qt5-qtsvg
+- qt5-plugin-imageformat-gif
+- qt5-qtsvg-plugin-imageformat-svg
+PkgBR:
+- qt5-qtdeclarative-import-location
+- qt5-qtdeclarative-import-positioning
```
Comment (veskuh) [inline rpm/tweetian.yaml]: imports are runtime dependencies, having these in Requires is enough

File: main.cpp (hunk 1 above)
Comment (siteshwar) [inline main.cpp:19]: This should do https://github.com/siteshwar/Tweetian/commit/7e029e1bccbab3d275ce905c2b242ca0b223545f

**Comment by veskuh** [comment]
Works on emulator, nice! Will try on device next :-)

**Comment by veskuh** [comment]
Ok, there is still stuff to do (fix icon, cleanup code, launching from icon, etc.), but in any case this is a great start. Merging.

File: src/thumbnailcacher.cpp [hunk 3]
```diff
@@ -70,22 +70,25 @@ QString ThumbnailCacher::get(const QString &id)
...
         return;
-
+/*
+    //TODO: Find the right way to cache an image
     QImage thumb(imageObj->boundingRect().size().toSize(), QImage::Format_ARGB32);
     thumb.fill(QColor(0,0,0,0).rgba());
     QPainter painter(&thumb);
     QStyleOptionGraphicsItem style;
-    imageObj->paint(&painter, &style, 0);
+    //imageObj->paint(&painter, &style, 0);
```
Comment (siteshwar) [inline src/thumbnailcacher.cpp:85]: @veskuh 
I couldn't find how to do this in Qt5. QQuickItem doesn't have a paint method. And the closest way I could find to do this is by taking a screenshot of full window with imageObj->window()->grabWindow() and crop thumbnail out of it. But I am not sure if it's the best way to do this. Your suggestions are welcome.

Comment (veskuh) [inline src/thumbnailcacher.cpp:85]: @siteshwar We do not need to port tweetian's own thumbnailer for Qt5, we can just use platform's thumbnailer. See https://github.com/nemomobile/nemo-qml-plugin-thumbnailer for details.

Comment (siteshwar) [inline src/thumbnailcacher.cpp:85]: I spent a few hours trying to understand how this plugin works, but couldn't get it to work. Can you give me a good example of how to use this plugin ?

Comment (veskuh) [inline src/thumbnailcacher.cpp:85]: You need to import it and then you just use url in style Image { .. source: "image://nemothumbnail/path/to/file/" }. See qmlgallery for example. https://github.com/nemomobile/qmlgallery/blob/master/qml/api/GalleryDelegate.qml

Comment (siteshwar) [inline src/thumbnailcacher.cpp:85]: This only seems to work for the images which are stored on local file system (and not for the images which are on a web server).

**Pull request merged**